catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -657,4 +571,126 @@ private Filter getTagsQueryFilter() {
                                 .attribute(Metacard.TAGS)
                                 .empty());
     }
+
+    static class QuerySources {
```

Comment:
Also, this class _almost_ implements the Builder pattern. Is it worth going all the way?